enterprise/server/workflow/service/service.go

```diff
@@ -592,18 +592,11 @@ func (ws *workflowService) getWorkflowByID(ctx context.Context, workflowID strin
 	return rwf.Workflow, nil
 }
 
-func (ws *workflowService) GetWorkflowByGroupAndRepo(ctx context.Context, groupID string, repoURL string) (*tables.Workflow, error) {
-	parsedRepoURL, err := gitutil.ParseGitHubRepoURL(repoURL)
-	if err != nil {
-		return nil, err
-	}
-
-	rwf, err := ws.getRepositoryWorkflow(ctx, groupID, parsedRepoURL)
-	if err != nil {
-		return nil, err
-	}
-
-	return rwf.Workflow, nil
+// GetLegacyWorkflowIDForGitRepository generates an artificial workflow ID so that legacy Workflow structs
+// can be created from GitRepositories to play nicely with the pre-existing architecture
+// that expects the legacy format
+func (ws *workflowService) GetLegacyWorkflowIDForGitRepository(groupID string, repoURL string) string {
```

Comment:
this function doesn't look like it needs to be defined on the struct anymore (which would let you go back to not having `gitRepositoryWorkflow` on the struct as well and eliminate a bunch of the diffs in here if I'm not misreading)

Reply:
The problem is that buildbuddy_server is in the OSS package, but workflow service is in the enterprise package. buildbuddy_server can't import the workflow file to use the static function, so defined the function on the struct

Reply:
oops! i missed the external call, thanks.